server/src/test/java/org/opensearch/bootstrap/BootstrapChecksTests.java

```diff
@@ -774,4 +779,22 @@ Version getVersion() {
         version.set(Runtime.version());
         BootstrapChecks.check(emptyContext, true, Collections.singletonList(check));
     }
+
+    public void testMultipleDataPathCheck() throws NodeValidationException {
+        Path path = PathUtils.get(createTempDir().toString());
+        String[] paths = new String[] { path.resolve("a").toString(), path.resolve("b").toString() };
+
+        final BootstrapContext context = createTestContext(
+            Settings.builder()
+                .putList(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), Collections.singletonList(DiscoveryNodeRole.SEARCH_ROLE.roleName()))
+                .putList(Environment.PATH_DATA_SETTING.getKey(), paths)
+                .build(),
+            Metadata.EMPTY_METADATA
+        );
+        final List<BootstrapCheck> checks = Collections.singletonList(new BootstrapChecks.MultipleDataPathCheck());
+        final NodeValidationException e = expectThrows(NodeValidationException.class, () -> BootstrapChecks.check(context, true, checks));
+        assertThat(e.getMessage(), containsString("Having multiple data paths in the search role is not allowed"));
```

Comment:
nit: `assertThat` -> `assertEquals`

Reply:
I think it is preferable to use `assertThat` due to https://github.com/opensearch-project/OpenSearch/pull/2503#pullrequestreview-914681694

Reply:
Isn't `assertThat` deprecated?

Reply:
`org.junit.Assert.assertThat()` is deprecated (which OpenSearchTestCase ultimately inherits from). `org.hamcrest.MatcherAssert.assertThat` is not deprecated and is the intended replacement for that.